File: swift/problem21.swift
public class ListNode {
    public var val: Int
    public var next: ListNode?
    public init(_ val: Int) {
        self.val = val
        self.next = nil
    }
}

class Solution {
    func mergeTwoLists(_ l1: ListNode?, _ l2: ListNode?) -> ListNode? {
        let ans = ListNode(0)
        var cur = ans; var p1 = l1; var p2 = l2
        
        while (p1 != nil && p2 != nil) {
            if (p1!.val < p2!.val) {
                cur.next = p1
                cur = p1!
                p1 = p1!.next
            } else {
                cur.next = p2
                cur = p2!
                p2 = p2!.next
            }
        }
        
        if (p1 != nil) {
            cur.next = p1
        } else if (p2 != nil) {
            cur.next = p2
        }
        
        return ans.next
    }
}

File: python/problem21.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        result=l=ListNode(0)  #2 listnode
        while l1 and l2:
            if l1.val<l2.val:
                l.next=l1
                l1=l1.next
                
            else:
                l.next=l2
                l2=l2.next
            l=l.next
        l.next=l1 or l2  
        return result.next
        